File: backend/app/schemas/water.py
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel, Field

DEFAULT_DAILY_TARGET_ML = 2000
# One glass, for the ml/glasses toggle on the history chart. Matches the
# smallest quick-add button in the UI.
GLASS_ML = 250


class WaterLogRequest(BaseModel):
    amount_ml: int = Field(ge=1, le=5000)
    log_date: date | None = None


class WaterLogResponse(BaseModel):
    id: uuid.UUID
    log_date: date
    amount_ml: int
    created_at: datetime

    model_config = {"from_attributes": True}


class WaterDailySummaryResponse(BaseModel):
    date: date
    total_ml: int
    goal_ml: int
    entries: list[WaterLogResponse]


class WaterHistoryDay(BaseModel):
    date: date
    day_label: str          # "Mon", "Tue", ...
    total_ml: int


class WaterHistoryResponse(BaseModel):
    start_date: date
    end_date: date
    days: list[WaterHistoryDay]
    total_ml: int
    average_ml: int
    daily_target_ml: int
    glass_ml: int


class WaterGoalRequest(BaseModel):
    daily_target_ml: int = Field(ge=100, le=10000)


class WaterGoalResponse(BaseModel):
    daily_target_ml: int

File: frontend/types/water.ts
export interface WaterEntry {
  id: string
  log_date: string        // YYYY-MM-DD
  amount_ml: number
  created_at: string
}

export interface WaterDailySummary {
  date: string             // YYYY-MM-DD
  total_ml: number
  goal_ml: number
  entries: WaterEntry[]
}

export interface WaterHistoryDay {
  date: string             // YYYY-MM-DD
  day_label: string        // "Mon", "Tue", ...
  total_ml: number
}

export interface WaterHistory {
  start_date: string
  end_date: string
  days: WaterHistoryDay[]
  total_ml: number
  average_ml: number
  daily_target_ml: number
  glass_ml: number
}

export interface WaterGoal {
  daily_target_ml: number
}

export interface LogWaterRequest {
  amount_ml: number
  log_date?: string        // YYYY-MM-DD, defaults to today server-side
}
